### ISSUE TITLE
The padding list in sliding_window_inference method is reversed

### ISSUE THREAD

**Issue by joaolcguerreiro** [Project-MONAI/MONAI#7126]
**Describe the bug**
The padding list (`pad_size`) in [sliding_window_inference](https://docs.monai.io/en/stable/_modules/monai/inferers/utils.html#sliding_window_inference) method is reversed.

**To Reproduce**
```
input_shape = [5, 3, 64, 64, 64]
roi_size = [66, 32, 32]
num_spatial_dims = len(input_shape) - 2
batch_size, _, *image_size_ = input_shape
image_size = tuple(max(image_size_[i], roi_size[i]) for i in range(num_spatial_dims))
pad_size = []
for k in range(len(input_shape) - 1, 1, -1):
    diff = max(roi_size[k - 2] - input_shape[k], 0)
    half = diff // 2
    pad_size.extend([half, diff - half])
print(pad_size)
```

`>>> [0, 0, 0, 0, 1, 1]`

**Expected behavior**
I believe the expected behaviour should be pad_size=[1, 1, 0, 0, 0, 0]

**Comment by joaolcguerreiro** [comment]
After looking at [SlidingWindowSplitter](https://docs.monai.io/en/latest/_modules/monai/inferers/splitter.html#SlidingWindowSplitter) I believe the current behaviour is intentional. What am I missing?

**Comment by joaolcguerreiro** [comment]
After looking at [SlidingWindowSplitter](https://docs.monai.io/en/latest/_modules/monai/inferers/splitter.html#SlidingWindowSplitter) I believe the current behaviour is intentional. What am I missing?

**Comment by wyli** [comment]
the pytorch api defines this param starting from the last dim https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html#torch.nn.functional.pad

**Comment by joaolcguerreiro** [comment]
Got it! Thank you 🙏
